src/components/Loader.vue

```diff
@@ -0,0 +1,342 @@
+ *
+ * The `state` is available as `Loader.Status.LOADING`, `Loader.Status.SUCCESS`,
+ * `Loader.Status.WARNING` and `Loader.Status.ERROR`.
+ *
+ * The events are available as `Loader.Events.MAIN_ACTION` and `Loader.Events.ALTERNATIVE_ACTION`.
+ */
+@Component
+class Loader extends Vue {
+    @Prop({type: String, default: 'Improving the world'}) private title!: string;
+    @Prop({type: String, default: Loader.Status.LOADING}) private state!: string;
+    @Prop(Boolean) private lightBlue?: boolean;
+    @Prop(String) private status?: string;
+    @Prop(String) private message?: string;
+    @Prop(String) private mainAction?: string;
+    @Prop(String) private alternativeAction?: string;
+
+    private currentStatus: string = '';
+    private nextStatus: string = '';
+    private isStatusTransitioning: boolean = false;
+
+    /**
+     * To enable a smooth transition of the non-transitionable background-image
+     * property, we instead place the new background above the old one and
+     * animate the top element's opacity. But because the color area has rounded
+     * corners, and the browser creates transparent pixels in the corner
+     * because of anti-aliasing, the blue background partly shines through the
+     * transparent corner pixels of the foreground. Thus we remove the background
+     * color after the transition is complete.
+     */
+    private showLoadingBackground: boolean = true;
+
+    private _loadingTitle?: string;
+
+    private get loadingTitle() {
+        return this._loadingTitle || (this._loadingTitle = this.title);
+    }
+
+    @Watch('state', {immediate: true})
+    private updateState(newState: string, oldState: string) {
+        // When the component is initialized with a state other than LOADING
+        if (!oldState && (newState !== Loader.Status.LOADING)) {
+            this.showLoadingBackground = false;
+        }
+
+        // When the state changes later and animates
+        if (oldState && (newState !== Loader.Status.LOADING)) {
+            setTimeout(() => this.showLoadingBackground = false, 1000);
+        }
+
+        if (newState === Loader.Status.LOADING) {
+            this.showLoadingBackground = true;
+        }
+    }
+
+    @Watch('status', {immediate: true})
+    private updateStatus(newStatus: string, oldStatus: string) {
+        this.nextStatus = newStatus;
+        this.isStatusTransitioning = true;
+
+        setTimeout(() => {
+            this.currentStatus = newStatus;
+            this.isStatusTransitioning = false;
+        }, 500);
+    }
+
+    private onMainAction() {
+        this.$emit(Loader.Events.MAIN_ACTION);
+    }
+
+    private onAlternativeAction() {
+        this.$emit(Loader.Events.ALTERNATIVE_ACTION);
+    }
+}
+
+namespace Loader { // tslint:disable-line no-namespace
+    export enum Status {
+        LOADING = 'loading',
+        SUCCESS = 'success',
+        WARNING = 'warning',
+        ERROR = 'error',
+    }
+
+    export enum Events {
```

Comment:
Singular or Plural.

Reply:
In the keyguard-next code we usually used `Events`

Reply:
Once the type is available it becomes somewhat strange as Singular for `enum`.
`const myCar: Cars = Cars.BlueCar;` does not read well.